httpx/dispatch/http2.py

```diff
@@ -107,7 +119,9 @@ def is_connection_dropped(self) -> bool:
         return self.socket.is_connection_dropped()
 
     async def close(self) -> None:
-        await self.socket.close()
+        async with self.write_lock:
+            async with self.read_lock:
```

Comment:
Alternatively…

```python
async with self.write_lock, self.read_lock:
    await self.socket.close()
```

Reply:
TIL

Reply:
Works with a regular `with:` and supports `as`, too!

```python
with open("file1.txt") as f1, open("file2.txt") as f2:
    ...
```

The more you know…